lib/Dialect/Verif/VerifOps.cpp

```diff
@@ -42,6 +54,46 @@ OpFoldResult HasBeenResetOp::fold(FoldAdaptor adaptor) {
   return {};
 }
 
+//===----------------------------------------------------------------------===//
+// AssertLikeOps Canonicalizations
+//===----------------------------------------------------------------------===//
+
+namespace {
+struct AssertLikeOp {
+public:
+  // Pattern:
+  // assertlike(ltl.clock(prop, clk), en) -> clocked_assertlike(prop, en, clk)
+  template <typename TargetOp, typename Op>
```

Comment:
nit: Can we remove outer struct if it's used as just a namespace? Just defining static function looks OK to me.